server/api/api/endpoints/projects_v2.py

```diff
@@ -63,6 +63,15 @@ async def delete_project(
         logger.info("Project not found, nothing to delete", project=name)
         return fastapi.Response(status_code=http.HTTPStatus.NO_CONTENT.value)
 
+    # usually the CRUD for delete project will check permissions, however, since we are running the crud in a background
+    # task, we need to check permissions here. skip permission check if the request is from the leader.
+    if not server.api.utils.helpers.is_request_from_leader(auth_info.projects_role):
```

Comment:
i was thinking about the boilerplate, I think worth shoving the `is_request_from_leader` inside the `query_project_permissions`

Reply:
That's not safe in my opinion. `query_project_permissions` isn't only called on projects CRUD, but on other sub resources as well, where this `if` is *not* boilerplated. If it were everywhere, I'd agree, but I wouldn't add this to all permission checks ever.